File: ReactP/reactp/shapeapi.py
import uuid
import random
from . import logger
from .toolshape import ToolShape

def RandomCircle():
    try:
        return random.randint(0, 255) < 25
    except Exception as e:
        logger.error("RandomCircleFail", exc_info=True)
        return False

def RandomColor():
    try:
        bytes_ = [random.randint(0, 255) for _ in range(3)]
        return "#{:02X}{:02X}{:02X}".format(*bytes_)
    except Exception as e:
        logger.error("RandomColorFail", exc_info=True)
        return ""

def RandomCoordinate(maximum):
    try:
        if maximum <= 0:
            return 0.0
        return random.random() * maximum
    except Exception as e:
        logger.error("RandomCoordinateFail", exc_info=True)
        return 0.0

def RandomOpacity(minimum, maximum):
    try:
        return minimum + (maximum - minimum) * random.random()
    except Exception as e:
        logger.error("RandomOpacityFail", exc_info=True)
        return 1.0

def RandomRotation():
    try:
        return random.random() * 360.0
    except Exception as e:
        logger.error("RandomRotationFail", exc_info=True)
        return 0.0

def RandomScale(minimum, maximum):
    try:
        return minimum + (maximum - minimum) * random.random()
    except Exception as e:
        logger.error("RandomScaleFail", exc_info=True)
        return 1.0

def RandomSides(minimum, maximum):
    try:
        if minimum > maximum:
            return 12
        return random.randint(minimum, maximum)
    except Exception as e:
        logger.error("RandomSidesFail", exc_info=True)
        return 12

def RandomStar():
    try:
        return random.randint(0, 255) < 50
    except Exception as e:
        logger.error("RandomStarFail", exc_info=True)
        return False

def generate_shape(input_shape: ToolShape = None) -> ToolShape:
    try:
        input_shape = input_shape or ToolShape()
        shape = ToolShape()
        shape.id = input_shape.id or uuid.uuid4()
        shape.x = input_shape.x if input_shape.x is not None else RandomCoordinate(7648)
        shape.y = input_shape.y if input_shape.y is not None else RandomCoordinate(4304)
        shape.circle = input_shape.circle if input_shape.circle is not None else RandomCircle()
        shape.opacity = input_shape.opacity if input_shape.opacity is not None else RandomOpacity(0.0, 1.0)
        shape.rgb = input_shape.rgb if input_shape.rgb is not None else RandomColor()
        shape.rotation = input_shape.rotation if input_shape.rotation is not None else RandomRotation()
        shape.scale = input_shape.scale if input_shape.scale is not None else RandomScale(0.1, 2.0)
        shape.sides = input_shape.sides if input_shape.sides is not None else RandomSides(3, 8)
        shape.star = input_shape.star if input_shape.star is not None else RandomStar()
        return shape
    except Exception as e:
        logger.error("GenerateShapeFail", exc_info=True)
        return None

File: ReactJ/src/main/java/KusDepot/ReactJ/ShapeAPI.java
package KusDepot.ReactJ;

import java.util.UUID;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import java.security.SecureRandom;

public final class ShapeAPI
{
    private static final SecureRandom rng = new SecureRandom();
    private static final Logger logger = LoggerFactory.getLogger(ShapeAPI.class);
    private ShapeAPI() {}

    public static ToolShape GenerateShape(ToolShape input)
    {
        try
        {
            ToolShape shape = new ToolShape();
            shape.setId(input != null && input.getId() != null ? input.getId() : UUID.randomUUID());
            shape.setX(input != null && input.getX() != null ? input.getX() : RandomCoordinate(7648));
            shape.setY(input != null && input.getY() != null ? input.getY() : RandomCoordinate(4304));
            shape.setCircle(input != null && input.getCircle() != null ? input.getCircle() : RandomCircle());
            shape.setOpacity(input != null && input.getOpacity() != null ? input.getOpacity() : RandomOpacity(0.0,1.0));
            shape.setRgb(input != null && input.getRgb() != null ? input.getRgb() : RandomColor());
            shape.setRotation(input != null && input.getRotation() != null ? input.getRotation() : RandomRotation());
            shape.setScale(input != null && input.getScale() != null ? input.getScale() : RandomScale(0.1,2.0));
            shape.setSides(input != null && input.getSides() != null ? input.getSides() : RandomSides(3,8));
            shape.setStar(input != null && input.getStar() != null ? input.getStar() : RandomStar());
            return shape;
        }
        catch ( Exception e )
        {
            logger.error("GenerateShapeFail",e); return null;
        }
    }

    private static boolean RandomCircle()
    {
        try
        {
            return rng.nextInt(256) < 25;
        }
        catch ( Exception e )
        {
            logger.error("RandomCircleFail",e); return false;
        }
    }

    private static String RandomColor()
    {
        try
        {
            byte[] bytes = new byte[3]; rng.nextBytes(bytes);

            return String.format("#%02X%02X%02X",bytes[0],bytes[1],bytes[2]);
        }
        catch ( Exception e )
        {
            logger.error("RandomColorFail",e); return "";
        }
    }

    private static double RandomCoordinate(int max)
    {
        try
        {
            if (max <= 0) { return 0; }

            return rng.nextDouble() * max;
        }
        catch ( Exception e )
        {
            logger.error("RandomCoordinateFail",e); return 0;
        }
    }

    private static double RandomOpacity(double min , double max)
    {
        try
        {
            return min + (max - min) * rng.nextDouble();
        }
        catch ( Exception e )
        {
            logger.error("RandomOpacityFail",e); return 1.0;
        }
    }

    private static double RandomRotation()
    {
        try
        {
            return rng.nextDouble() * 360.0;
        }
        catch ( Exception e )
        {
            logger.error("RandomRotationFail",e); return 0.0;
        }
    }

    private static double RandomScale(double min , double max)
    {
        try
        {
            return min + (max - min) * rng.nextDouble();
        }
        catch ( Exception e )
        {
            logger.error("RandomScaleFail",e); return 1.0;
        }
    }

    private static int RandomSides(int min , int max)
    {
        try
        {
            if (min > max) { return 12; }

            return rng.nextInt(max - min + 1) + min;
        }
        catch ( Exception e )
        {
            logger.error("RandomSidesFail",e); return 12;
        }
    }

    private static boolean RandomStar()
    {
        try
        {
            return rng.nextInt(256) < 50;
        }
        catch ( Exception e )
        {
            logger.error("RandomStarFail",e); return false;
        }
    }
}